### DISCUSSION TITLE
Issue with applied background color

### DISCUSSION THREAD

**Discussion by Pankaj003** [google/filament#5967] [Q&A]
Hi,

Need help regarding few issues related to background color:
This is related to https://github.com/google/filament/issues/5898#issue-1331670294

I am trying to set background color of the 3D object by following calls:
filament::LinearColor linearCorr = filament::Color::toLinearfilament::ACCURATE({ color.redF(), color.greenF(), color.blueF() });
RenderingEngine::Get().ClearOptions().clearColor = filament::math::float4({ linearCorr [0], linearCorr [1], linearCorr [2], color.alphaF() });
RenderingEngine::Get().ClearOptions().clear = true;
RenderingEngine::Get().Renderer()->setClearOptions(RenderingEngine::Get().ClearOptions());
	
The issue is, the color which I am selecting is not getting applied but some shade of it.
for example: if I chose rgb(142, 177, 85) then on checking the Window color using color picker tool I am getting value rgb(208, 213, 178).
The color can be verified by small color image on the color selection dialog.

Interesting part is, if I query  Renderer::ClearOptions::clearColor object then it shows rgb(142, 177, 85), the original intended color.

Looks like while rendering, the color is getting diluted, may be due to lighting setting or something else.

Is there a way I can either get to apply the intended color or can get the changed color value from some filament api?
This I am facing on both Windows and Mac OS.

Please check attached screen shot.

![ColorIssue](https://user-images.githubusercontent.com/8086656/183398843-5159dcc9-b9ab-4ed2-9e47-ec1b41230bce.PNG)

**Comment by romainguy** [comment]
You need to use linear tone mapping or to apply tbe inverse tone mapping curve to your color.

**Reply by romainguy** [reply]
See https://github.com/google/filament/blob/357000a0e7f0b7f52338c008e30de5b7dda8458c/libs/viewer/src/Settings.cpp#L536
